crypto/asn1/a_time.c

```diff
@@ -441,6 +400,31 @@ int ASN1_TIME_diff(int *pday, int *psec,
     return OPENSSL_gmtime_diff(pday, psec, &tm_from, &tm_to);
 }
 
+int ASN1_TIME_cmp_time_t(const ASN1_TIME *s, time_t t)
+{
+    struct tm stm, ttm;
+    int day, sec;
+
+    if (!ASN1_TIME_to_tm(s, &stm))
+        return -2;
+
+    if (!OPENSSL_gmtime(&t, &ttm))
+        return -2;
+
+    if (!OPENSSL_gmtime_diff(&day, &sec, &ttm, &stm))
+        return -2;
+
+    if (day > 0)
```

Comment:
Apologies for the ambiguity.  I meant there was no documentation for `OPENSLL_gmtime_diff` to guarantee that days and seconds returned both must have the same sign.  The source does guarantee this, so assuming it seems reasonable.

`ASN1_TIME_diff` has such a statement in the POD.